### PR TITLE
Use smaller and compressed varients of buttons and form components

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description
Replace instances of `EuiButton` that don't have an explicit sizing attribute to `EuiSmallButton*`.
Replace instances of `Eui<form elements>` that don't have density attributes to `EuiCompressed<form elements>`.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/main/loading_modal/loading_modal.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
   EuiFlexGroup,
   EuiFlexItem,
   EuiLoadingSpinner,
-  EuiButton,
```
Comment (derek-ho) [inline public/components/main/loading_modal/loading_modal.tsx:17]: Should this just be removed, seems not used?

File: public/components/main/reports_table.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import React, { Fragment, useState } from 'react';
 import { i18n } from '@osd/i18n';
 import {
-  EuiButton,
+  EuiSmallButton,
```
Comment (derek-ho) [inline public/components/main/reports_table.tsx:9]: Same comment as above

**Review by derek-ho — APPROVED** [review]

**Comment by derek-ho** [comment]
Looks like snapshots need to be updated as well

**Review by derek-ho — APPROVED** [review]

File: public/components/main/loading_modal/loading_modal.tsx (hunk 1 above)
Comment (AMoo-Miki) [reply public/components/main/loading_modal/loading_modal.tsx:17]: yep; not used.

**Comment by AMoo-Miki** [comment]
The failing tests are due to changes that will be coming in OUI 1.8.2. Let's leave this open and try retry when OSD gets updated to OUI 1.8.2.

**Comment by AMoo-Miki** [comment]
OUI 1.9 was merged into main and the only failures are for to linting issues unrelated to this PR.

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**